main.go

```diff
@@ -0,0 +1,185 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/google/go-github/v48/github"
+	"golang.org/x/oauth2"
+)
+
+var appDataOrg = "redhat-appstudio-appdata"
+
+func main() {
+	if os.Getenv("GITHUB_TOKEN") == "" {
+		fmt.Println("GITHUB_TOKEN must be set as an environment variable")
+	}
+	githubToken := os.Getenv("GITHUB_TOKEN")
+
+	// Initialize an authenticated github client
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: githubToken},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+	authClient := github.NewClient(tc)
+
+	// Parse command line flags to determine which operation to perform
+	// Options are:
+	// Delete repository matching a given name
+	// Delete all invalid repositories (ones starting with a dash)
+	// Listing all repositories
+	var operation, keyword, repo string
+	flag.StringVar(&operation, "operation", "", "The operation to perform. One of: delete-repo or delete-invalid")
+	flag.StringVar(&keyword, "keyword", "", "The keyword(s) to match gitops repositories on")
+	flag.StringVar(&repo, "repo", "", "The name of a repository")
+	flag.Parse()
+
+	// Check the values of the flags before proceding
+	if operation == "" {
+		log.Fatal("usage: --operation must be set as a command-line flag")
+	}
+
+	if operation != "delete-invalid" && operation != "list-all" && operation != "delete-repo" {
+		log.Fatal("usage: The only valid options for '--operation' are delete-repo, delete-invalid, or list-all")
+	}
+
+	if operation == "delete-invalid" {
+		invalidRepos, err := listInvalidRepos(ctx, authClient)
+		if err != nil {
+			log.Fatal(err.Error())
+		}
+		err = deleteRepos(ctx, authClient, invalidRepos)
+		if err != nil {
+			log.Fatal(err.Error())
+		}
+	} else if operation == "list-all" {
+		allRepos, err := listAllRepos(ctx, authClient)
+		if err != nil {
+			log.Fatal(err.Error())
+		}
+		for _, repo := range allRepos {
+			fmt.Println(*repo.Name)
+		}
+
+	} else {
+		if repo == "" {
+			log.Fatal("usage: --repo <repo-name> must be passed in as a flag when using the 'delete' operation")
+		}
+		err := deleteRepo(ctx, authClient, repo)
+		if err != nil {
+			log.Fatal(err.Error())
+		}
+	}
+
+}
+
+// listAllRepos returns the list of all github repositories in the org
+func listAllRepos(ctx context.Context, client *github.Client) ([]*github.Repository, error) {
+	opt := &github.RepositoryListByOrgOptions{
+		ListOptions: github.ListOptions{PerPage: 100},
+	}
+	var allRepos []*github.Repository
+	for {
+		repos, resp, err := client.Repositories.ListByOrg(ctx, appDataOrg, opt)
+		if err != nil {
+			return nil, err
+		}
+		allRepos = append(allRepos, repos...)
+		if resp.NextPage == 0 {
+			break
+		}
+		opt.Page = resp.NextPage
+	}
+	return allRepos, nil
+
+}
+
+// listInvalidRepos return any gitops repositories in the org starting with '-'
```

Comment:
I'll be honest, I hadn't thought that far ahead. Right now, that's the only scenario that comes to mind. 

The bug in HAS allowed Application resources to be created without a displayName, since HAS requires the displayName field to be set to successfully reconcile Applications, it would basically get stuck creating endless GitOps repositories (with the name missing, hence the starting `-`).

The bug has since been fixed, but I opted to keep this in in case there's other scenarios in the future that lead to "invalid repositories" like we identified here.